framework/src/utils/SlepcSupport.C

```diff
@@ -361,144 +403,185 @@ setWhichEigenPairsOptions(SolverParams & solver_params)
+          dont_add_these_options, "-init_eps_power_snes_mf_operator", "1");
   }
   else
   {
-    Moose::PetscSupport::setSinglePetscOption("-snes_mf_operator", "0");
+    Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+        dont_add_these_options, "-snes_mf_operator", "0");
     if (initial_power)
-      Moose::PetscSupport::setSinglePetscOption("-init_eps_power_snes_mf_operator", "0");
+      Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+          dont_add_these_options, "-init_eps_power_snes_mf_operator", "0");
   }
 #if PETSC_RELEASE_LESS_THAN(3, 13, 0)
-  Moose::PetscSupport::setSinglePetscOption("-st_type", "sinvert");
+  Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+      dont_add_these_options, "-st_type", "sinvert");
   if (initial_power)
-    Moose::PetscSupport::setSinglePetscOption("-init_st_type", "sinvert");
+    Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+        dont_add_these_options, "-init_st_type", "sinvert");
 #endif
 #else
   mooseError("Newton-based eigenvalue solver requires SLEPc 3.7.3 or higher");
 #endif
 }
 
 void
-setNonlinearPowerOptions(SolverParams & solver_params)
+setNonlinearPowerOptions(SolverParams & solver_params, EigenProblem & eigen_problem)
 {
+  const auto & dont_add_these_options = eigen_problem.getPetscOptions().dont_add_these_options;
 #if !SLEPC_VERSION_LESS_THAN(3, 8, 0) || !PETSC_VERSION_RELEASE
-  Moose::PetscSupport::setSinglePetscOption("-eps_type", "power");
-  Moose::PetscSupport::setSinglePetscOption("-eps_power_nonlinear", "1");
-  Moose::PetscSupport::setSinglePetscOption("-eps_target_magnitude", "");
+  Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+      dont_add_these_options, "-eps_type", "power");
+  Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+      dont_add_these_options, "-eps_power_nonlinear", "1");
+  Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+      dont_add_these_options, "-eps_target_magnitude", "");
   if (solver_params._eigen_matrix_free)
-    Moose::PetscSupport::setSinglePetscOption("-snes_mf_operator", "1");
+    Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+        dont_add_these_options, "-snes_mf_operator", "1");
   else
-    Moose::PetscSupport::setSinglePetscOption("-snes_mf_operator", "0");
+    Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+        dont_add_these_options, "-snes_mf_operator", "0");
 
 #if PETSC_RELEASE_LESS_THAN(3, 13, 0)
-  Moose::PetscSupport::setSinglePetscOption("-st_type", "sinvert");
+  Moose::PetscSupport::setSinglePetscOptionIfAppropriate(
+      dont_add_these_options, "-st_type", "sinvert");
 #endif
 #else
   mooseError("Nonlinear Inverse Power requires SLEPc 3.7.3 or higher");
 #endif
 }
 
 void
-setEigenSolverOptions(SolverParams & solver_params, const InputParameters & params)
+setEigenSolverOptions(SolverParams & solver_params,
+                      const InputParameters & params,
+                      EigenProblem & eigen_problem)
```

Comment:
same question. Like why would we not set `-eps_type power` when the `solver_params._eigen_solve_type` is `Moose::EST_POWER`?